dartagnan/src/main/java/com/dat3m/dartagnan/wmm/RelationNameRepository.java

```diff
@@ -19,34 +19,23 @@ public class RelationNameRepository {
     public static final String ADDRDIRECT = "addrDirect";
     public static final String CTRLDIRECT = "ctrlDirect";
     public static final String EMPTY = "0";
-    public static final String FR = "fr";
     public static final String IDDTRANS = "idd^+";
     public static final String DATA = "data";
     public static final String ADDR = "addr";
     public static final String CTRL = "ctrl";
-    public static final String POLOC = "po-loc";
-    public static final String RFE = "rfe";
-    public static final String RFI = "rfi";
-    public static final String COE = "coe";
-    public static final String COI = "coi";
-    public static final String FRE = "fre";
-    public static final String FRI = "fri";
     public static final String MFENCE = "mfence";
     public static final String ISH = "ish";
     public static final String ISB = "isb";
     public static final String SYNC = "sync";
     public static final String ISYNC = "isync";
     public static final String LWSYNC = "lwsync";
```

Comment:
Can we somehow even get rid of these fence definitions? It's unfortunate that the relation has the same name as the event tag, so we probably cannot define `let mfence = po;[mfence];po`.
Well, maybe it works (I haven't tried).

Reply:
I considered moving each definition to `X.fences.cat` and get rid of those, but then some of the parsers were complaining. Maybe I can move the string to each particular parser (similar to how `RefinementSolver` defines the strings it needs)

Reply:
I see. The `RefinementSolver` thing needs to be solved differently anyways because the biases will only work if the user has defined all those relations (which might not be the case!). Unless that is intended, but then we need to generate proper error messages if the relations are missing.

Reply:
How hard would it be to take a second cat model as input for the bias?

Reply:
Probably not too hard.

Reply:
I got rid of all fence definitions in the code.

I leave it to you to fix the `RefinementSolver` :P

Reply:
Can we define `fencerel(r)` rather than having special handling for it?

Reply:
https://github.com/hernanponcedeleon/Dat3M/pull/736/commits/3b8c0dfaeaa1f5a57385dafd433ebd17dfe67474 best commit ever

Reply:
Yep, very nice. I thought you would just define `let fencerel(r) = po;[r];po` though. But I don't mind either way.

Reply:
> I leave it to you to fix the `RefinementSolver` :P

Ok, this is not super straightforward due to cutting. I would rather not fix this in this PR and let the biases be only semi-functional for now.

Reply:
Actually, I don't think it will work at all because the relations are queried on the baseline model (which is empty most of the time). So the feature might be non-functional.

Reply:
> I would rather not fix this in this PR and let the biases be only semi-functional for now.

Do you at least have an idea of who this should be fixed?

Reply:
I think we need the functionality to intersect/merge two memory models. Then we can generate a Wmm by cutting the original one and then merge into this one the bias model.

Reply:
Ok. Is there anything missing for the PR or can I merge once the CI is green?

Reply:
I think nothing is missing.

Reply:
As I said above, the original memory model also needs to get cut in many cases and the cut part needs to be part of the baseline model. So you need to be able to merged two models (the cut model and the parsed baseline model).

Reply:
> the original memory model also needs to get cut 

Is this something that is only needed after the changes in this PR because, what needs to be cut, before used to be considered base relations?

Reply:
No, originally we first cut the model and on the resulting model we then programatically added the biases. For the later part we relied on `getPredefinedRelation` for a few cases like `fr` and `po-loc`. I guess the construction of those few relations could be moved into the bias-generating code though.

Reply:
I did this and now biases seem to be functional. Unless you have some objections about my last commit, I think we can finally merge

Reply:
I think it's fine. Technically, `fr` gets created even if it is not needed in some cases, but I guess that shouldn't matter much.